corehq/apps/users/views/mobile/users.py

```diff
@@ -1301,47 +1321,86 @@ def _excel_data(self, rows):
     else:
         return HttpResponseBadRequest("Invalid Request")
+
     user_count = 0
-    for domain in user_filters['domains']:
-        user_count += get_commcare_users_by_filters(domain, user_filters, count_only=True)
+    (is_multi_domain_download, domains_list) = get_domains_from_user_filters(domain, user_filters)
+    for current_domain in domains_list:
+        if user_type == MOBILE_USER_TYPE:
+            user_count += count_mobile_users_by_filters(current_domain, user_filters)
+        else:
+            user_count += count_web_users_by_filters(current_domain, user_filters)
+            user_count += count_invitations_by_filters(current_domain, user_filters)
+
     return json_response({
         'count': user_count
     })
 
 
 @require_can_edit_or_view_commcare_users
 def download_commcare_users(request, domain):
-    form = CommCareUserFilterForm(request.GET, domain=domain, couch_user=request.couch_user)
+    return download_users(request, domain, user_type=MOBILE_USER_TYPE)
+
+
+@login_and_domain_required
+def download_users(request, domain, user_type):
+    if user_type not in [MOBILE_USER_TYPE, WEB_USER_TYPE]:
+        raise AssertionError(f"Invalid user type for download_users: {user_type}")
+
+    form = UserFilterForm(request.GET, domain=domain, couch_user=request.couch_user, user_type=user_type)
     if form.is_valid():
         user_filters = form.cleaned_data
     else:
-        return HttpResponseRedirect(
-            reverse(FilteredUserDownload.urlname, args=[domain]) + "?" + request.GET.urlencode())
+        view = FilteredCommCareUserDownload if user_type == MOBILE_USER_TYPE else FilteredWebUserDownload
+        return HttpResponseRedirect(reverse(view, args=[domain]) + "?" + request.GET.urlencode())
     download = DownloadBase()
     if form.cleaned_data['domains'] != [domain]:  # if additional domains added for download
-        track_workflow(request.couch_user.username, 'Domain filter used for mobile download')
-    is_web_download = False
-    if form.cleaned_data['columns'] == CommCareUserFilterForm.USERNAMES_COLUMN_OPTION:
+        user_type = 'mobile' if user_type == MOBILE_USER_TYPE else 'web'
+        track_workflow(request.couch_user.username, f'Domain filter used for {user_type} download')
+    if form.cleaned_data['columns'] == UserFilterForm.USERNAMES_COLUMN_OPTION:
+        if user_type != MOBILE_USER_TYPE:
+            raise AssertionError("USERNAME_COLUMN_OPTION only available for mobile users")
         res = bulk_download_usernames_async.delay(domain, download.download_id, user_filters,
                                                   owner_id=request.couch_user.get_id)
     else:
         res = bulk_download_users_async.delay(domain, download.download_id, user_filters,
-                                              is_web_download, owner_id=request.couch_user.get_id)
+                                              (user_type == WEB_USER_TYPE), owner_id=request.couch_user.get_id)
```

Comment:
nit: I personally would favor having `is_web_download = user_type == WEB_USER_TYPE` earlier and then passing it in argument, or just pass user_type and let the called method handle it.

Reply:
Let's agree to disagree on this one. I don't want to make a named variable earlier in the function since it's only used in this one place, and I'm also not keen to change the signature of `bulk_download_users_async` in this PR.